src/types/message.rs

```diff
@@ -794,6 +795,29 @@ mod getters {
     }
 }
 
+impl Message {
+    pub fn url(&self) -> Option<String> {
+        match &self.chat.kind {
+            ChatKind::NonPrivate {
+                kind:
+                    NonPrivateChatKind::Channel {
+                        username: Some(username),
+                    },
+                ..
+            }
+            | ChatKind::NonPrivate {
+                kind:
+                    NonPrivateChatKind::Supergroup {
+                        username: Some(username),
+                        ..
+                    },
+                ..
+            } => Some(format!("https://t.me/{0}/{1}/", username, self.id)),
```

Comment:
How about using https://docs.rs/reqwest/0.10.1/reqwest/struct.Url.html ?

Reply:
I think this is excess

Reply:
Why you don't want to use types? It'll still be `Display`, but also provide more functionality.

Reply:
fixed

Reply:
fixed